### PR TITLE
feat(death): allow customization of ignored regions

### LINKED ISSUE #414: Feature Request: Option to ignore deaths in specific regionIDs
### Checklist

- [X] I've searched the [issues and pull requests](https://github.com/pajlads/Dinkplugin/issues?q=) for similar looking suggestions.
- [X] I've checked the `Unreleased` section of the [changelog](https://github.com/pajlads/DinkPlugin/blob/master/CHANGELOG.md) for newly added features that sound like my suggestion.

### Describe your Suggestion

UIMs love to deathpile in specific spots, mostly in the middle of priff. It'd be cool if they could disable sending death screenshots when they're just deathpiling.

### Reasoning

It's quite spammy when you get multiple of these per day from each UIM:

afbeelding

They're not really fun to look at, since it's a very intentional death; not someone fucking up.

### PR DESCRIPTION
**Pull request by iProdigy**
Closes #414

### CODE REVIEW

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 1]
```diff
@@ -943,12 +943,25 @@ default Set<ExceptionalDeath> deathSafeExceptions() {
...
+        keyName = "deathIgnoredRegions",
+        name = "Ignored Region IDs",
+        description = "User-specified, comma-separated Region IDs where deaths should be ignored.<br/>" +
+            "Use the '::DinkRegion' chat command or an online map to obtain the region IDs.<br/>" +
+            "For example, Prifddinas spans 12894, 12895, 13150, 13151",
+        position = 45,
+        section = deathSection
+    )
+    default String deathIgnoredRegions() {
+        return "8781, 12336, 13462, 13463";
```
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPluginConfig.java:956]: I feel like we can use ones that are just in the middle of nowhere? e.g. in the ocean, most of the time you can't even access those so you should be unable to die in them.

```suggestion
        return "4904, 5930, 6953, 7723";
```

<details>
<summary>locations</summary>

image

</details>

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 2]
```diff
@@ -943,12 +943,25 @@ default Set<ExceptionalDeath> deathSafeExceptions() {
         return EnumSet.of(ExceptionalDeath.FIGHT_CAVE, ExceptionalDeath.INFERNO);
     }
 
+    @ConfigItem(
+        keyName = "deathIgnoredRegions",
+        name = "Ignored Region IDs",
+        description = "User-specified, comma-separated Region IDs where deaths should be ignored.<br/>" +
+            "Use the '::DinkRegion' chat command or an online map to obtain the region IDs.<br/>" +
```
Comment (Felanbird) [inline src/main/java/dinkplugin/DinkPluginConfig.java:950]:
```suggestion
            "Use the '::dinkregion' chat command or an online map to obtain the region IDs.<br/>" +
```
command is case-insensitive and our export command readme mentions are lowercase

**Review by Felanbird — CHANGES_REQUESTED** [review]
nits

File: src/main/java/dinkplugin/DinkPluginConfig.java [hunk 3]
```diff
@@ -943,12 +943,25 @@ default Set<ExceptionalDeath> deathSafeExceptions() {
...
+        keyName = "deathIgnoredRegions",
+        name = "Ignored Region IDs",
+        description = "User-specified, comma-separated Region IDs where deaths should be ignored.<br/>" +
+            "Use the '::dinkregion' chat command or an online map to obtain the region IDs.<br/>" +
+            "For example, Prifddinas spans 12894, 12895, 13150, 13151",
+        position = 45,
+        section = deathSection
+    )
+    default String deathIgnoredRegions() {
+        return "4904, 5930, 6953, 7723";
```
Comment (iProdigy) [inline src/main/java/dinkplugin/DinkPluginConfig.java:956]: perhaps we use some obviously fake IDs so its clearer that users can delete them

```suggestion
        return "123, 456, 789";
```

**Review by Felanbird — APPROVED** [review]
I tested the region where the Jail Guard is outside of Draynor `12338` and it worked as expected

**Review by pajlada — COMMENTED** [review]
I would like this feature, together with the `::DinkRegion` command, mentioned in the readme file, but that's probably best to do once #416 has been merged in. The text should mention deathpiling & UIM so it can be CTRL+F'd easily

**Review by pajlada — APPROVED** [review]

**Pull request merged**